x-pack/plugins/serverless_search/public/application/components/connectors/edit_service_type.tsx

```diff
@@ -70,15 +156,41 @@ export const EditServiceType: React.FC<EditServiceTypeProps> = ({ connector, isD
       data-test-subj="serverlessSearchEditConnectorType"
       fullWidth
     >
-      <EuiSuperSelect
+      <EuiComboBox
+        aria-label={i18n.translate(
+          'xpack.serverlessSearch.connectors.chooseConnectorSelectable.euiComboBox.accessibleScreenReaderLabelLabel',
+          { defaultMessage: 'Select a data source for your connector to use.' }
+        )}
         // We only want to allow people to set the service type once to avoid weird conflicts
-        disabled={Boolean(connector.service_type) || isDisabled}
-        data-test-subj="serverlessSearchEditConnectorTypeChoices"
+        isDisabled={Boolean(connector.service_type) || isDisabled}
         isLoading={isLoading}
-        onChange={(event) => mutate(event)}
-        options={options}
-        valueOfSelected={connector.service_type || undefined}
+        data-test-subj="serverlessSearchEditConnectorTypeChoices"
+        prepend={
+          <EuiIcon
+            type={
+              connector.service_type
+                ? connectorTypes.find((conn) => conn.serviceType === connector.service_type)
+                    ?.iconPath ?? ''
+                : `${assetBasePath}/connectors.svg`
+            }
+            size="l"
+          />
+        }
+        singleSelection
         fullWidth
+        placeholder={i18n.translate(
+          'xpack.serverlessSearch.connectors.chooseConnectorSelectable.placeholder.text',
+          { defaultMessage: 'Choose a data source' }
+        )}
+        options={initialOptions}
+        selectedOptions={initialOptions.filter(
+          (option) => option.serviceType === connector.service_type
```

Comment:
It's not clear to me how to get a unique serviceType from here. I'm getting 3 items for any Jira connector selected. @sphilipse how could I apply your suggestion `array.filter((item,index) => array.indexOf(item) === index));` here?

![CleanShot 2024-11-13 at 15 20 55@2x](https://github.com/user-attachments/assets/c75e82e0-c0d5-4540-b35f-d1b2421c4803)

Reply:
If so, we should just display 1 Jira item in the dropdown IMO not 3. That would be more accurate and less confusing. 

![CleanShot 2024-11-13 at 15 41 40@2x](https://github.com/user-attachments/assets/fe76a521-d425-41a4-9050-bc1bf652094a)

cc: @danajuratoni @seanstory 

Currently I realised this serviceType issue is also affecting the current experience in ES3 if you watch the video choosing any _Confluence_ (either Cloud & Server or Data Center) connector always applies the _Confluence Cloud_ & Server in the table and the same happens choosing any _Jira_ (either Cloud, Server or Data Center)  connector it always applies the _Jira Cloud_


https://github.com/user-attachments/assets/548fc89d-0718-4df9-b8af-b4b6d5708567

Reply:
> If so, we should just display 1 Jira item in the dropdown IMO not 3. That would be more accurate and less confusing.

I'm fine with that. I think that's how we did confluence and github in ESS, where you pick which "flavor" you want in the configuration tab, not from the tiles at creation time.

Reply:
This is what we display now in ESS for Confluence:

![CleanShot 2024-11-14 at 11 46 10@2x](https://github.com/user-attachments/assets/d56efaf7-5e33-4f23-b1c9-18120ed9e472)

And this for Github:

![CleanShot 2024-11-14 at 11 47 25@2x](https://github.com/user-attachments/assets/bb7bfd59-5493-49d5-a554-06be70378d40)